runtime/domain.c

```diff
@@ -1533,22 +1544,16 @@ static void caml_poll_gc_work(void)
 /* FIXME: do not raise async exceptions */
 void caml_handle_gc_interrupt(void)
 {
-  atomic_uintnat* young_limit = domain_self->interruptor.interrupt_word;
   CAMLalloc_point_here;
 
   CAML_EV_BEGIN(EV_INTERRUPT_GC);
-  if (atomic_load_acq(young_limit) == INTERRUPT_MAGIC) {
-    /* interrupt */
+  if (caml_incoming_interrupts_queued()) {
```

Comment:
If I understand correctly, there is a change of behavior here. Before, the test would check for any service interrupt, now we only check for STW interrupts. This makes sense because the call to `caml_handle_incoming_interrupts` within the conditional, if I understand correctly, only handle STW interrupts anyway.

Reply:
Right, except there is no change of behaviour for the reason you explain.